### PR TITLE
[3.5] Add UnsafeReadScheduledCompact and UnsafeReadFinishedCompact

### PR DESCRIPTION
**Pull request by CaojiamingAlan**
Follow up https://github.com/etcd-io/etcd/pull/16068#discussion_r1263667496

Please read https://github.com/etcd-io/etcd/blob/main/CONTRIBUTING.md#contribution-flow.

### CODE REVIEW

File: server/mvcc/util.go [hunk 1]
```diff
@@ -41,3 +41,19 @@ func UnsafeSetScheduledCompact(tx backend.BatchTx, value int64) {
 	revToBytes(revision{main: value}, rbytes)
 	tx.UnsafePut(buckets.Meta, scheduledCompactKeyName, rbytes)
 }
+
+func UnsafeReadFinishedCompact(tx backend.ReadTx) (finishedCompact int64, found bool) {
```
Comment (ahrtr) [inline server/mvcc/util.go:45]:
```suggestion
func UnsafeReadFinishedCompact(tx backend.ReadTx) (int64, bool) {
```

File: server/mvcc/util.go [hunk 2]
```diff
@@ -41,3 +41,19 @@ func UnsafeSetScheduledCompact(tx backend.BatchTx, value int64) {
...
+	_, finishedCompactBytes := tx.UnsafeRange(buckets.Meta, finishedCompactKeyName, nil, 0)
+	if len(finishedCompactBytes) != 0 {
+		return bytesToRev(finishedCompactBytes[0]).main, true
+	}
+	return 0, false
+}
+
+func UnsafeReadScheduledCompact(tx backend.ReadTx) (scheduledCompact int64, found bool) {
```
Comment (ahrtr) [inline server/mvcc/util.go:53]:
```suggestion
func UnsafeReadScheduledCompact(tx backend.ReadTx) (int64, bool) {
```

**Review by ahrtr — APPROVED** [review]

**Review by fuweid — APPROVED** [review]
LGTM

**Review by jmhbnz — APPROVED** [review]
LGTM - Thanks for completing this follow-up @CaojiamingAlan 👍🏻

**Pull request merged**